### ISSUE TITLE
[Bug] sopa-v1.0.14 errors on Xenium v2 + multimodal cell segmentation data but not dapi-only data

### ISSUE THREAD

**Issue by sooheelee** [gustaveroussy/sopa#80]
Bonjour. This is my first time running sopa. I found both sopa-v1.0.14 and v1.0.13 errors on Xenium v2 pipeline data that includes multimodal cell segmentation stains but not on v2 pipeline data that uses only DAPI. The error is
```
ValueError: cannot reindex or align along dimension 'c' because of conflicting dimension sizes: {16, 4}
```
Full stack-trace: [sopa-1.0.14-read-error-20240607.txt](https://github.com/user-attachments/files/15745968/sopa-1.0.14-read-error-20240607.txt)

Using the DAPI-only v2 data succeeds and gives the stdout:
```
SpatialData object with:
├── Images
│     └── 'morphology_focus': MultiscaleSpatialImage[cyx] (1, 3553, 5791), (1, 1776, 2895), (1, 888, 1447), (1, 444, 723), (1, 222, 361)
└── Points
      └── 'transcripts': DataFrame with shape: (<Delayed>, 11) (3D points)
with coordinate systems:
▸ 'global', with elements:
        morphology_focus (Images), transcripts (Points)
```

Both data sets are public, small and at https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/resources/xenium-example-data.

- the Breast Gene Expression data uses multimodal cell segmentation (3 stains + DAPI) https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_human_Breast_2fov/Xenium_V1_human_Breast_2fov_outs.zip
- the Lung Gene Expression data uses nuclear expansion cell segmentation (DAPI only) https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_human_Lung_2fov/Xenium_V1_human_Lung_2fov_outs.zip

Command was `sopa read Xenium_V1_human_Lung_2fov_outs  --technology xenium --sdata-path sopa` for the latter. 

Given the discussion in https://github.com/gustaveroussy/sopa/issues/68, expected `read` function to work. May be I am missing something? Thanks in advance for your help.

 - OS: macOS Ventura 13.6.3 
 - Python 3.9.13

**Comment by quentinblampey** [comment]
Hello @sooheelee, thanks for the detailed issue. I'm aware of this issue, and it has been fixed in `spatialdata-io`, but I'm waiting for a release of `spatialdata-io` to have a specific PR (https://github.com/scverse/spatialdata-io/pull/151) available to have it inside sopa. It should come in one or a few weeks, I'll let you know!

**Comment by sooheelee** [comment]
Thank you.

**Comment by littleju777** [comment]
Hi, I also met the same issue with multimodal cell segmentation data. I noticed that the tutorials were mostly wrote for Xenium with DAPI data, do you find methods work perfectly with multimodal data? Maybe using Xenium results + baysor would be better for Xenium V2.0 data. Do you have any updates? Thank you!

**Comment by quentinblampey** [comment]
Hi @littleju777, I haven't checked the multimodal segmentation yet. But indeed it may be better to use the Xenium results as a prior for Baysor, I will try to add this!

**Comment by littleju777** [comment]
> Hi @littleju777, I haven't checked the multimodal segmentation yet. But indeed it may be better to use the Xenium results as a prior for Baysor, I will try to add this!

Thank you!

**Comment by quentinblampey** [comment]
Hi @sooheelee, I released `sopa==1.1.1`, which should fix this. Can you try and let me know?

And @littleju777 you can now use the multimodal segmentation as a prior for baysor. I provided a [baysor config](https://github.com/gustaveroussy/sopa/blob/master/workflow/config/xenium/baysor_multimodal.yaml) for snakemake, or you can also look at the [FAQ here](https://gustaveroussy.github.io/sopa/faq/#how-to-use-a-prior-cell-segmentation) if you prefer using the CLI/API.

**Comment by littleju777** [comment]
> Hi @sooheelee, I released `sopa==1.1.1`, which should fix this. Can you try and let me know?
> 
> And @littleju777 you can now use the multimodal segmentation as a prior for baysor. I provided a [baysor config](https://github.com/gustaveroussy/sopa/blob/master/workflow/config/xenium/baysor_multimodal.yaml) for snakemake, or you can also look at the [FAQ here](https://gustaveroussy.github.io/sopa/faq/#how-to-use-a-prior-cell-segmentation) if you prefer using the CLI/API.

Thank you for the amazing work!
